app/frontend/components/module_assistant/assistant_checkbox.js

```diff
@@ -0,0 +1,212 @@
+    font-size: 0.8em;
+    opacity: 0.8;
+    margin-bottom: -0.5em;
+    margin-inline-start: 2.2em;
+    line-height: 1em;
+  }
+  #info {
+    display: none;
+    margin-left: 0.5em;
+  }
+  #info svg {
+    height: 1.2em;
+    fill: var(--accent-color, gray);
+  }
+</style>
+ 
+<div id="label-wrapper">
+  <label>  
+    <input type="checkbox">
+    <slot name="label-icon"></slot>
+    <slot name="label-text">No text provided</slot>
+    <span id="count" class="empty"></span>
+  </label>
+  <a id="info" href="#">${ICON_INFO}</a>
+</div>
+<div id="description"></div>
+`;
+
+class AssistantCheckbox extends HTMLElement {
+  static get observedAttributes() {
+    return ['count', 'description', 'checked'];
+  }
+
+  constructor() {
+    super();
+    this.attachShadow({ mode: 'open' });
+    this.shadowRoot.appendChild(template.content.cloneNode(true));
+
+    this._checked = false;
+    this._disabled = false;
+    this.code = "";
+
+    this.checkbox = this.shadowRoot.querySelector('input[type="checkbox"]');
+    this.checkbox.addEventListener('change', () => this.handleCheckboxChecked());
+  }
+
+  connectedCallback() {  
+    this.code = this.getAttribute('code');
+    
+    this._disabled = this.hasAttribute('disabled');
+    if (this._disabled) {
+      this.checkbox.setAttribute('disabled', '');
+      this.shadowRoot.querySelector('label').classList.add('disabled');
+    }
+
+    if (this.hasAttribute('info')) {
+      const infoElement = this.shadowRoot.querySelector('#info');
+      const infoHint = this.getAttribute('info');
+      if (infoHint !== '') {
+        infoElement.setAttribute('title', infoHint);
+      }
+
+      infoElement.style.display = 'block';
+      infoElement.addEventListener('click', () => this.handleInfoClick());
+    }
+  }
+
+  attributeChangedCallback(name, oldValue, newValue) {
+    if (name === 'checked') {
+      this.handleCheckedAttr(oldValue, newValue);
+      return;
+    }
+
+    if (name === 'count' && newValue) {
+      newValue = ` (${newValue})`;
+    }
+
+    this.update(name, newValue);
+  }
+
+  set count(n) {
+    if (n) {
+      this.setAttribute('count', n);
+    } else {
+      this.removeAttribute('count');
+    }
+  }
+
+  set description(text) {
+    if (text) {
+      this.setAttribute('description', text);
+    } else {
+      this.removeAttribute('description');
+    }
+  }
+
+  set checked(isChecked) {
+    if (isChecked) {
+      this.setAttribute('checked', '');
+    } else {
+      this.removeAttribute('checked');
+    }
+  }
+
+  handleCheckedAttr(oldValue, newValue) {
+    this._checked = newValue !== null;
+    this.checkbox.checked = this._checked;
+  }
+
+  handleCheckboxChecked() {
+    if (this._disabled) {
+      return;
+    }
+
+    this._checked = this.checkbox.checked;
+    if (this._checked) {
+      this.setAttribute('checked', '');
+    } else {
+      this.removeAttribute('checked');
+    }
+
+    this.dispatchEvent(new CustomEvent("itemSelected", {
```

Comment:
`itemSelected` only matches the usecase when you select, but you can also deselect, so I think something more neutral would make more sense. I think I would opt for `change`, since that is also used by standard HTML elements.

Reply:
The catching of this event is done by step web components (it bubules up). So I will opt for unique name so random input or checkboxes would not trigger the handler. `itemChanged`